sdk/azurestackhci/Azure.ResourceManager.StackHCI/src/Generated/Extension.cs

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Core;
+using Azure.ResourceManager.StackHCI.Models;
+using Azure.ResourceManager.Resources.Models;
+
+namespace Azure.ResourceManager.StackHCI
+{
+    /// <summary> A Class representing a Extension along with the instance operations that can be performed on it. </summary>
+    public partial class Extension : ArmResource
```

Comment:
Avoid using sing-word class names and use a more specific name with directive settings in autorest.md.

Reply:
This is an arm-resource present in the swagger. I'm not sure if we can rename this. I could only see directives worked for files under Models.

Reply:
Use a directive like:
```
- from: filename.json
  where: $.definitions.Extension
  transform: $["x-ms-client-name"] = "XXXExtension"
```
Please also change for `Cluster`.

Reply:
Done